asma/core/sources/github.py

```
@@ -143,6 +148,20 @@ def resolve(self, skill: Skill) -> ResolvedSource:
                 # Use version as tag
                 ref = skill.version
         else:
+            # No version or ref specified
+            if self.strict:
+                raise ValueError(
+                    f"Version not specified for skill '{skill.name}'. "
+                    f"In strict mode, you must specify 'version' or 'ref' in skillset.yaml."
+                )
+
+            # Emit warning and use default branch
+            warnings.warn(
```

Ruff (0.14.10)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)